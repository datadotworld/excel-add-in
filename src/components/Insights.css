/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
.insights-container {
  background-color: white;
  overflow: scroll;
}

.insight-header {
  padding: 15px;
  width: 100%;
}

.insight-title {
  font-size: 18px;
  font-weight: 700;
}

.close-button {
  float: right;
  cursor: pointer;
}

.insight-charts {
  width: 80%;
  margin: 0 auto;
  padding-bottom: 30px;
}
