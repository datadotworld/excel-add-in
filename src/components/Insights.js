/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid, Row } from 'react-bootstrap';
import Charts from './Charts';
import UploadInsight from './UploadInsight';

import './Insights.css';

class Insights extends Component {
  static propTypes = {
    user: PropTypes.object,
    getImageAndTitle: PropTypes.func,
    charts: PropTypes.array,
    officeInitialized: PropTypes.bool,
    projects: PropTypes.array,
    createProject: PropTypes.func,
    uploadChart: PropTypes.func,
    initializeInsights: PropTypes.func
  };

  state = {
    selectedChart: ''
  };

  selectChart = (imageString, title) => {
    this.setState({ selectedChart: { imageString, title } });
  };

  render() {
    const { selectedChart } = this.state;
    const {
      charts,
      getImageAndTitle,
      uploadChart,
      projects,
      user,
      createProject,
      officeInitialized,
      setError
    } = this.props;

    if (officeInitialized) {
      return (
        <Grid className="insights-container">
          <Row className="center-block section-header insight-header">
            <div className="insight-title">New Insight</div>
          </Row>
          {!selectedChart && (
            <Charts
              charts={charts}
              projects={projects}
              getImageAndTitle={getImageAndTitle}
              user={user}
              createProject={createProject}
              selectChart={this.selectChart}
              setError={setError}
            />
          )}
          {selectedChart && (
            <UploadInsight
              getImageAndTitle={getImageAndTitle}
              chart={selectedChart.imageString}
              title={selectedChart.title}
              uploadChart={uploadChart}
              projects={projects}
              setError={this.props.setError}
            />
          )}
        </Grid>
      );
    } else {
      return <div />;
    }
  }
}

export default Insights;
