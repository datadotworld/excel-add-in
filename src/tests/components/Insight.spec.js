/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import Insights from '../../components/Insights';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Insights charts={[]} projects={[]}/>, div);
});

it('renders charts view - no charts', () => {
  expect(renderer.create(<Insights
    charts={[]}
    projects={['project 1', 'project 2']}
    officeInitialized="true"
    />
  ).toJSON()).toMatchSnapshot()
});

it('renders charts view - chart', () => {
  const charts = [
    { sheet: '{00000000-0001-0000-0000-000000000000}', chartName: 'Chart 1' },
    { sheet: '{00000000-0001-0000-0000-000000000000}', chartName: 'Chart 2' }
  ];

  const getImageAndTitle = () => {
    return new Promise((resolve, reject) => {
      resolve({ image: 'b64string', title: 'A Title' });
    })
  }

  expect(renderer.create(
    <IntlProvider locale='en'>
      <Insights
        charts={charts}
        getImageAndTitle={getImageAndTitle}
        projects={[]}
        officeInitialized="true"
      />
    </IntlProvider>).toJSON()).toMatchSnapshot()
});

it('renders no projects view', () => {
  expect(renderer.create(<Insights charts={[]} projects={[]} />).toJSON()).toMatchSnapshot()
});