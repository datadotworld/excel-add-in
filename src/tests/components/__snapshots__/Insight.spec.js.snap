// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders charts view - chart 1`] = `
<div
  className="insights-container container"
>
  <div
    className="center-block section-header insight-header row"
  >
    <div
      className="insight-title"
    >
      New Insight
    </div>
  </div>
  <div
    className="charts-container row"
  >
    <div
      className="no-chart"
    >
      <div>
        Create a project for your insight.
      </div>
      <button
        className="no-chart-button btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Create Project
      </button>
      <div>
        <button
          className="no-chart-button btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Refresh
        </button>
      </div>
    </div>
  </div>
  
</div>
`;

exports[`renders charts view - no charts 1`] = `
<div
  className="insights-container container"
>
  <div
    className="center-block section-header insight-header row"
  >
    <div
      className="insight-title"
    >
      New Insight
    </div>
  </div>
  <div
    className="charts-container row"
  >
    <div
      className="no-chart"
    >
      <div>
        Insert a chart to get started.
      </div>
      <button
        className="no-chart-button btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
  
</div>
`;

exports[`renders no projects view 1`] = `<div />`;
